packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -38,53 +38,36 @@ abstract class UnpackMacOS extends Target {
   ];
 
   @override
-  List<Source> get outputs => const <Source>[];
+  List<Source> get outputs => const <Source>[
+    Source.pattern('{OUTPUT_DIR}/FlutterMacOS.framework/FlutterMacOS'),
+  ];
 
   @override
   List<Target> get dependencies => <Target>[];
 
-  @override
-  List<String> get depfiles => const <String>['unpack_macos.d'];
-
   @override
   Future<void> build(Environment environment) async {
     if (environment.defines[kBuildMode] == null) {
       throw MissingDefineException(kBuildMode, 'unpack_macos');
     }
     final BuildMode buildMode = getBuildModeForName(environment.defines[kBuildMode]);
     final String basePath = environment.artifacts.getArtifactPath(Artifact.flutterMacOSFramework, mode: buildMode);
-    final Directory targetDirectory = environment
-      .outputDir
-      .childDirectory('FlutterMacOS.framework');
-    // Deleting this directory is required or else the FlutterMacOS module
-    // cannot be found.
-    if (targetDirectory.existsSync()) {
-      targetDirectory.deleteSync(recursive: true);
-    }
-    final List<File> inputs = environment.fileSystem.directory(basePath)
-      .listSync(recursive: true)
-      .whereType<File>()
-      .toList();
-    final List<File> outputs = inputs.map((File file) {
-      final String relativePath = environment.fileSystem.path.relative(file.path, from: basePath);
-      return environment.fileSystem.file(environment.fileSystem.path.join(targetDirectory.path, relativePath));
-    }).toList();
-    final ProcessResult result = await environment.processManager
-        .run(<String>['cp', '-R', basePath, targetDirectory.path]);
+
+    final ProcessResult result = environment.processManager.runSync(<String>[
+      'rsync',
+      '-av',
+      '--delete',
+      '--filter',
+      '- .DS_Store/',
+      basePath,
+      environment.outputDir.path,
+    ]);
```

Comment:
Please update the comment on this class, which currently says we're using `cp -R`.